In [36]:
import numpy as np
import pandas as pd
import tensorflow as tf
from sklearn.model_selection import train_test_split

In [37]:
df = pd.read_csv('winequality-red.csv')
df.head()

fixed acidity  volatile acidity  citric acid  residual sugar  chlorides  \
0            7.4              0.70         0.00             1.9      0.076   
1            7.8              0.88         0.00             2.6      0.098   
2            7.8              0.76         0.04             2.3      0.092   
3           11.2              0.28         0.56             1.9      0.075   
4            7.4              0.70         0.00             1.9      0.076   

   free sulfur dioxide  total sulfur dioxide  density    pH  sulphates  \
0                 11.0                  34.0   0.9978  3.51       0.56   
1                 25.0                  67.0   0.9968  3.20       0.68   
2                 15.0                  54.0   0.9970  3.26       0.65   
3                 17.0                  60.0   0.9980  3.16       0.58   
4                 11.0                  34.0   0.9978  3.51       0.56   

   alcohol  quality  
0      9.4        5  
1      9.8        5  
2      9.8        5  
3      9.8        6  
4      9.4        5

In [38]:
X = df.drop('quality', axis=1)
y = df['quality']

In [39]:

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

In [40]:
model = tf.keras.models.Sequential([
    tf.keras.layers.Dense(128, activation='relu', input_shape=[11]),
    tf.keras.layers.Dense(128, activation='relu'),
    tf.keras.layers.Dense(1)
])
model.compile(optimizer='adam', loss='mse', metrics=['mae'])

In [41]:
model.fit(X_train, y_train,
	validation_data=(X_test, y_test),
	epochs=10
)

Epoch 1/10
40/40 [==============================] - 1s 12ms/step - loss: 8.4498 - mae: 1.9567 - val_loss: 1.0624 - val_mae: 0.7611
Epoch 2/10
40/40 [==============================] - 0s 8ms/step - loss: 0.8012 - mae: 0.7029 - val_loss: 0.6768 - val_mae: 0.6450
Epoch 3/10
40/40 [==============================] - 0s 7ms/step - loss: 0.5815 - mae: 0.5941 - val_loss: 0.4942 - val_mae: 0.5465
Epoch 4/10
40/40 [==============================] - 0s 9ms/step - loss: 0.5146 - mae: 0.5536 - val_loss: 0.4982 - val_mae: 0.5425
Epoch 5/10
40/40 [==============================] - 0s 7ms/step - loss: 0.5310 - mae: 0.5689 - val_loss: 0.5306 - val_mae: 0.6018
Epoch 6/10
40/40 [==============================] - 0s 7ms/step - loss: 0.5266 - mae: 0.5653 - val_loss: 0.4701 - val_mae: 0.5407
Epoch 7/10
40/40 [==============================] - 0s 6ms/step - loss: 0.5088 - mae: 0.5507 - val_loss: 0.4814 - val_mae: 0.5485
Epoch 8/10
40/40 [==============================] - 0s 7ms/step - loss: 0.5269 - mae: 0.5

MAE = 0.54, which means quality is ±1, which is a pretty good model.